dal/elastic.go

```diff
@@ -2,126 +2,109 @@ package dal
 
 import (
 	"context"
+	"encoding/json"
 	"fmt"
+	"github.com/YuvalSteiy/book_service/models"
 	"github.com/olivere/elastic/v7"
-	"log"
-	"strconv"
 )
-const INDEXNAME = "books_yuval"
-// Book struct for deserialization of documents from json
-type Book struct {
-	Title          string  `json:"title"`
-	AuthorName     string  `json:"author_name"`
-	EbookAvailable bool    `json:"ebook_available"`
-	Price          float64 `json:"price"`
-	PublishDate    string  `json:"publish_date"`
+
+const INDEX_NAME = "books_yuval"
+const PORT = 9200
+const DOMAIN_URL = "http://es-search-7.fiverrdev.com"
+
+type Elastic struct {
+	Client *elastic.Client
 }
 
-func ElasticInitClient() *elastic.Client {
-	// Pass connection port
-	port := 9200
-	strPort := strconv.Itoa(port)
-	// Pass domain string
-	domainURL := "http://es-search-7.fiverrdev.com"
-	// Complete URL
-	setURL := domainURL + ":" + strPort
-	// Instantiate a client (elastic library instance)
+func (e *Elastic) InitClient() error {
```

Comment:
don't attach this function to your `Elastic` structure - the `Elastic` object has no right to exist without the client filed

create a contractor that  returns an `Elastic` object that contain an init `client` field